apps/web/src/app/vault/individual-vault/vault-onboarding/services/vault-onboarding.service.ts

```diff
@@ -16,23 +16,23 @@ export type VaultOnboardingTasks = {
   installExtension: boolean;
 };
 
-const VAULT_ONBOARDING_KEY = new KeyDefinition<any>(VAULT_ONBOARDING, "data", {
+const VAULT_ONBOARDING_KEY = new KeyDefinition<VaultOnboardingTasks>(VAULT_ONBOARDING, "tasks", {
   deserializer: (jsonData) => jsonData,
 });
 
 @Injectable()
 export class VaultOnboardingService implements VaultOnboardingServiceAbstraction {
-  private vaultOnboardingState: ActiveUserState<Record<string, VaultOnboardingTasks>>;
-  vaultOnboardingState$: Observable<Record<string, VaultOnboardingTasks>>;
+  private vaultOnboardingState: ActiveUserState<VaultOnboardingTasks>;
+  vaultOnboardingState$: Observable<VaultOnboardingTasks>;
 
   constructor(private stateProvider: StateProvider) {
     this.vaultOnboardingState = this.stateProvider.getActive(VAULT_ONBOARDING_KEY);
     this.vaultOnboardingState$ = this.vaultOnboardingState.state$;
   }
```

Comment:
nit (non-blocking): We can move the assignment of this state and observable to its declaration and out of the constructor to keep things together and is how we've done the other state provider migrations in Vault.

```suggestion
  private vaultOnboardingState = this.stateProvider.getActive(VAULT_ONBOARDING_KEY);
  vaultOnboardingState$ = this.vaultOnboardingState.state$;
  
  constructor(private stateProvider: StateProvider) { }
```

Reply:
Thanks. I'll note this and make the change in the follow-up onboarding M2 work